data/data/kubevirt/bootstrap/main.tf

```diff
@@ -35,18 +35,100 @@ resource "kubernetes_secret" "bootstrap_ignition" {
 }
 
 resource "kubevirt_virtual_machine" "bootstrap_vm" {
-  wait = true
 
-  name                 = "${var.cluster_id}-bootstrap"
-  namespace            = var.namespace
-  labels               = var.labels
-  storage_size         = var.storage
-  memory               = var.memory
-  cpu                  = var.cpu
-  storage_class_name   = var.storage_class
-  network_name         = var.network_name
-  access_mode          = var.pv_access_mode
-  ignition_secret_name = kubernetes_secret.bootstrap_ignition.metadata[0].name
-  image_url            = var.image_url
-  // pvc_name = var.pvc_name
+  metadata {
+    name      = "${var.cluster_id}-bootstrap"
+    namespace = var.namespace
+    labels    = var.labels
+  }
+  spec {
+    run_strategy = "Always"
+    data_volume_templates {
+      metadata {
+        name      = "${var.cluster_id}-bootstrap-bootvolume"
+        namespace = var.namespace
+      }
+      spec {
+        source {
+          http {
+            url = var.image_url
+          }
+        }
+        pvc {
+          access_modes = [var.pv_access_mode]
+          resources {
+            requests = {
+              storage = var.storage
+            }
+          }
+          storage_class_name = var.storage_class
+        }
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          "kubevirt.io/vm" = "${var.cluster_id}-bootstrap"
+        }
+      }
+      spec {
+        volumes {
```

Comment:
Shouldn't 'volumes' needs to be from array type and then 'volume' objects in that list?

Reply:
Here I'm using nested block syntax (attribute as block):
https://www.terraform.io/docs/configuration/attr-as-blocks.html

Can be more than one block, it used as array

Reply:
fine, so each block name should be singular and not plural

Reply:
Changed - https://github.com/nirarg/installer/pull/4/commits/1911a77f0238b3dcbf2e3f88b02bf988eb28a314